wallet-new/src/Cardano/Wallet/Kernel/DB/Read.hs

```diff
@@ -55,11 +68,37 @@ accountAvailableUtxo :: DB -> HdAccountId -> Utxo
 accountAvailableUtxo snapshot accountId
     = walletQuery' snapshot (Spec.queryAccountAvailableUtxo accountId)
 
+-- | Returns the available balance for the input 'HdAccountId'.
+accountAvailableBalance :: DB -> HdAccountId -> Coin
+accountAvailableBalance snapshot accountId
+    = walletQuery' snapshot (Spec.queryAccountAvailableBalance accountId)
+
 -- | Returns the total balance for this 'HdAccountId'.
 accountTotalBalance :: DB -> HdAccountId -> Coin
 accountTotalBalance snapshot accountId
     = walletQuery' snapshot (Spec.queryAccountTotalBalance accountId)
 
+-- | Returns the total balance for this 'HdAccountId'.
+accountAddresses :: DB -> HdAccountId -> IxSet HdAddress
+accountAddresses snapshot accountId
+    = walletQuery' snapshot (readAddressesByAccountId accountId)
+
 -- | Returns the total balance for this 'HdAccountId'.
 hdWallets :: DB -> HdWallets
 hdWallets snapshot = snapshot ^. dbHdWallets
+
+-- | Reads the given 'Address' in the 'HdAccount' current checkpoint.
+-- It relies on the 'Monoid' instance for an 'AddressMeta' in case such
+-- 'AddressMeta' cannot be found.
+readAddressMeta :: DB -> HdAccountId -> Address -> AddressMeta
```

Comment:
Let's replace this by a lens `getAddressMeta :: Address -> Lens' BlockMeta AddressMeta` (using the `lens` operator `none`, IIRC) so that this logic that "address absent means false false" gets encoded in the code rather than comments.